Hunting Queries/OfficeActivity/TeamsFilesUploaded.yaml

```diff
@@ -0,0 +1,31 @@
+id: 90e198a9-efb6-4719-ad89-81b8e93633a7
+name: Files uploaded to teams and access summary
+description: |
+  'Provides a summary of files uploaded to teams chats and extracts 
+  the users and IP addresses that have accessed them.'
+requiredDataConnectors:
+  - connectorId: Office365
+    dataTypes:
+      - OfficeActivity
+tactics:
+  - InitialAccess
+  - Exfiltration
+relevantTechniques:
+  - T1199
+  - T1102
+  - T1078
+query: |
+
+  OfficeActivity 
+  | where RecordType =~ "SharePointFileOperation" 
+  | where SourceRelativeUrl contains "Microsoft Teams Chat Files" 
+  | where Operation =~ "FileUploaded" 
+  | join kind= leftouter ( 
+     OfficeActivity 
+      | where RecordType =~ "SharePointFileOperation" 
+      | where SourceRelativeUrl contains "Microsoft Teams Chat Files" 
+      | where Operation =~ "FileDownloaded" or Operation =~ "FileAccessed" 
+  ) on OfficeObjectId 
+  | extend userBag = pack(UserId1, ClientIP1) 
+  | summarize makeset(UserId1), make_bag(userBag) by TimeGenerated, UserId, OfficeObjectId, SourceFileName 
```

Comment:
You could also add:
| extend NumberUsers = array_length(bag_keys(bag_userBag))
This gives a count of the number of users who have accessed the file, you can then project that and allow users to filter results by this number.

Reply:
Great addition, I have added this in!